kvbc/src/v4blockchain/detail/blockchain.cpp

```diff
@@ -197,8 +202,16 @@ std::optional<BlockId> Blockchain::loadLastReachableBlockId() {
   return concordUtils::fromBigEndianBuffer<BlockId>(itr.keyView().data());
 }
 
-// get the closest key to INITIAL_GENESIS_BLOCK_ID
+// if the genesis_block_key is found it means that we had pruning and we saved the genesis id in storage
+// in order to save iteration over a deleted range that takes time.
+// if it wasn't found we get the closest value to INITIAL_GENESIS_BLOCK_ID.
 std::optional<BlockId> Blockchain::loadGenesisBlockId() {
+  auto opt_val = native_client_->get(v4blockchain::detail::MISC_CF, kvbc::keyTypes::genesis_block_key);
```

Comment:
Maybe you already do it somewhere else - should we have a sanity check to see that this information is valid?
Try to see that for the blockId fetched from genesis_block_key, ID-1 is not there? This should take no time and makes sure we don't have some DB corruption.

Reply:
will add it, thanks!